### ISSUE TITLE
Holesky: State download starts after genesis

### ISSUE THREAD

**Issue by etan-status** [ethereum/go-ethereum#28219]
#### System information

Geth version: `1.13.2-stable`
Git Commit: dc34fe8291bfcaefbce97f559e9610beffb2e470
Architecture: amd64
Go Version: go1.21.1
Operating System: darwin
GOPATH=
GOROOT=

CL client & version: Nimbus v23.9.1
OS & Version: macOS 14.0 Sonoma (23A344)

#### Expected behaviour

Genesis should work

#### Actual behaviour

- State download via snap sync is started only after the chain is already operating.
- `Beacon backfilling failed                err="retrieved hash chain is invalid: missing parent"` errors are spammed every slot
- Nimbus reports optimistically synced / Geth doesn't reply `VALID` on engine

#### Steps to reproduce the behaviour

1. Clear Geth and Nimbus data directories
2. `geth --holesky --log.vmodule='rpc=5' --state.scheme path`
3. `build/nimbus_beacon_node --data-dir="$HOME/Downloads/nimbus/data/holesky" --network=holesky --rest --tcp-port=9010 --udp-port=9010 --history=prune --el=http://127.0.0.1:8551 --jwt-secret=/Users/etan/Library/Ethereum/holesky/geth/jwtsecret --log-level=DEBUG`

#### Backtrace
[holesky-geth-fail.txt](https://github.com/ethereum/go-ethereum/files/12749247/holesky-geth-fail.txt)

**Comment by fjl** [comment]
Did this issue keep persisting? We don't know which head block was being requested in the engine API. From Geth's perspective, the beacon client is requesting sync to a header that isn't in the chain.

**Comment by fjl** [comment]
I suspect this is an issue related to hash computation mismatch between CL and EL. Please provide additional information / dumps of the engine API requests.

**Comment by holiman** [comment]
This (and another issue) was opened just as holesky launched, snapsyncing during launch. It seems that the sync fails, state-sync completes but the chain-sync does not, and maybe the final "trigger" to complete the sync never happens, so the snapsync doesn't finish. 

We could probably reproduce this locally, if someone wants to spend an hour or two setting up a similar environment.
